aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -0,0 +1,208 @@
+#include "ATen/Context.h"
+#include "ATen/cuda/CUDAContext.h"
+#include "ATen/Dispatch.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/cuda/PinnedMemoryAllocator.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+
+#include "ATen/native/LinearAlgebraUtils.h"
+#include "ATen/native/cuda/MiscUtils.h"
+
+#include "THC.h" // for USE_MAGMA
+
+#ifdef USE_MAGMA
+#include <magma.h>
+#include <magma_types.h>
+#endif
+
+namespace at {
+namespace native {
+
+#ifdef USE_MAGMA
+template<class scalar_t>
+void magmaGesvBatched(
+    magma_int_t n, magma_int_t nrhs, scalar_t** dA_array, magma_int_t ldda,
+    magma_int_t** dipiv_array, scalar_t** dB_array, magma_int_t lddb,
+    magma_int_t* dinfo_array, magma_int_t batch_count, MAGMAQueue magma_queue) {
```

Comment:
These should be pass by const reference now (`const MAGMAQueue&`) so we don't copy the entire data structure when it is passed into these functions

Reply:
Understood, my bad. I'll fix it.